Goals fix

Denpeer opened the pull request:
Goals are now working properly.
Before this pull they'd always exit.
The tests now take a lot longer since the agent never exits (this is because the current agent doesn't build more buildings than the indicator target).

Denpeer committed:
Changed the way goals are completed

Denpeer committed:
removed debug predicates [ci skip]

Denpeer committed:
Goals are now adopted following the GOAL coding conventions

Denpeer committed:
Removed unnecessary action in DUWO (main module)

Denpeer committed:
Added order description to modules for clarification
Note: this doesn't change anything in the way the module is executed by
goal.

Denpeer committed:
Added order description to modules for clarification
Note: this doesn't change anything in the way the module is executed by
goal.

Denpeer committed:
Merge branch 'goalsFix' of https://github.com/Denpeer/VirtualHumansContext into goalsFix

Conflicts:
	src/DUWOagent/lowBudgetMod.mod2g
	src/DUWOagent/noBudgetMod.mod2g
	src/DUWOagent/normalMod.mod2g

klikooo commented on src/DUWOagent/events.mod2g:
```diff
@@ -43,4 +44,15 @@ module events {
+		
+	% Adopt goal to raise budget when our budget is below the target.
+	if bel(noBudget), not(goal(goalReachBudgetTarget))
+		then adopt(goalReachBudgetTarget).
```
Can this be placed in a seperate module?

Denpeer replied:
@klikooo This is the events module, I don't thing you're supposed to divide it into seperate modules.

klikooo replied:
Nick (TA from mas project) told me this is possible and is good coding practice

Denpeer replied:
Allright, I'll add it

MaxGroenenboom commented on src/DUWOagent/lowBudgetMod.mod2g:
```diff
@@ -1,6 +1,8 @@
 use predicates as knowledge.
 
+
+order=linearall.
```
Shouldn't this be `order=linear`?

Denpeer replied:
No, it should be `order=linear.` fixed it in a new commit.

Denpeer committed:
Refactored event.mod2g into different modules on recommendation of a mas project TA

Denpeer committed:
Reverted change in actions.act2g
this was for a different function, so reverting this and commiting it in another branch will keep the git clean :)

Denpeer committed:
Updated tests with the new modules from 4521fd3

Denpeer committed:
Modified goalEvents and perceptEvents to use linearall as order.
This is the default setting for the event module (but not for seperate modules)

klikooo commented on src/DUWOagent/events.mod2g:
```diff
@@ -1,46 +1,14 @@
-			do insert(indicator(ID,CurrentValue,TargetValue)).
+	% Adopting goals
+	if true then goalEvents.
+
```
Does it reach this, because the order is linear if I am right only the perceptEvents module is executed?

Denpeer replied:
I think event.mod2g is standard linearall. I'll change it to say that for clarity though.

MaxGroenenboom commented:
Looks good, I approve ;)

Denpeer committed:
added `order=linearall.` to `events.mod2g` for clarity

MarcianoJ commented on src/DUWOagent/events.mod2g:
```diff
@@ -2,6 +2,9 @@ use predicates as knowledge.
 use goalEvents as module.
 
+
+order=linear.
```
I think you meant to write linearall here

Denpeer replied:
:flushed:

Denpeer committed:
`4b25e30` fix

klikooo commented on src/DUWOagent/perceptEvents.mod2g:
```diff
@@ -0,0 +1,49 @@
+	
+	%% Delete all current indicators when we percept new indicators.
+	forall percept(indicators(_)), bel(indicator(ID,CurrentValue,TargetValue))
+		do delete(indicator(ID,CurrentValue,TargetValue)).
```
Add a tab here for consistency

Denpeer replied:
:fist: :point_down: :+1: :pray:

Denpeer committed:
Tab added for consistency

MarcianoJ commented:
Didn't notice the missing tab :P, but after that I approve
